src/libs/tagging.c

```diff
@@ -3418,37 +3421,34 @@ static void _lib_tagging_tag_show(dt_action_t *action)
   }
 
   d->floating_tag_imgs = dt_act_on_get_images(FALSE, TRUE, FALSE);
-  gint px = 0, py =0;
   GtkWidget *window = dt_ui_main_window(darktable.gui->ui);
   GtkWidget *center = dt_ui_center(darktable.gui->ui);
-  gdk_window_get_origin(gtk_widget_get_window(center), &px, &py);
-
-  const gint w = gdk_window_get_width(gtk_widget_get_window(center));
-  const gint h = gdk_window_get_height(gtk_widget_get_window(center));
-
-  const gint x = px + 0.5 * (w - FLOATING_ENTRY_WIDTH);
-  const gint y = py + h - 50;
-
-
-  /* put the floating box at the mouse pointer */
-  //   gint pointerx, pointery;
-  //   GdkDevice *device =
-  //   gdk_device_manager_get_client_pointer(gdk_display_get_device_manager(gtk_widget_get_display(widget)));
-  //   gdk_window_get_device_position (gtk_widget_get_window (widget), device, &pointerx, &pointery, NULL);
-  //   x = px + pointerx + 1;
-  //   y = py + pointery + 1;
+#ifdef GDK_WINDOWING_WAYLAND
```

Comment:
Why not simply?

`const gboolean on_wayland = dt_gui_get_session_type() == DT_GUI_SESSION_WAYLAND;`